scrnaseq_hto.Rmd

```diff
@@ -324,34 +559,137 @@ p = Seurat::DimPlot(sc, reduction="umap", group.by="hash.ID", cols=hto_colours$c
 p
 ```
 
+* Note: HTO colors are the same before and after filtering
+
+### Before filtering (each hto)
+```{r umap_cluster_before, fig.height=sc_all_fig_height}
+p = Seurat::DimPlot(sc.all, reduction="umap", group.by="seurat_clusters", split.by="hash.ID", ncol=2, cols=sc_all_col_clusters) + 
+  AddStyle(title="UMAP for each HTO, including doublets and negatives", 
+           legend_title="Cluster", 
+           legend_position="right")
+p
+```
+
+* Important: cluster colors are not neccessarily the same before and after filtering!
+
+### After filtering (each hto)
+```{r umap_cluster_after, fig.height=sc_fig_height}
+p = Seurat::DimPlot(sc, reduction="umap", group.by="seurat_clusters", split.by="hash.ID", ncol=2, cols=sc_col_clusters) + 
+  AddStyle(title="UMAP for each HTO, singlets only", 
+           legend_title="Cluster", 
+           legend_position="right")
+p
+```
+
+* Important: cluster colors are not neccessarily the same before and after filtering!
+
+## Distribution of cells in HTO and in clusters
+
+Next, we examine how the HTOs are distributed in the individual clusters and how the clusters are distributed in each individual HTO. We include negatives and doublets as they often group into specific clusters.
+
+```{r hto_cluster distribution}
+
+p1 = ggplot(sc.all[[c("seurat_clusters","hash.ID")]], aes(x=seurat_clusters, fill=hash.ID)) +
+  geom_bar(position="fill", colour="black") +
+  scale_x_discrete("Cluster") +
+  scale_y_continuous("Fraction of cells") +
+  scale_fill_manual(values=hto_colours$col_hto_collapsed) +
+  AddStyle(title="HTO cells in cluster", 
+           legend_title="Cell classification", 
+           legend_position="right")  
+
+p2 = ggplot(sc.all[[c("seurat_clusters","hash.ID")]], aes(x=hash.ID, fill=seurat_clusters)) +
+  geom_bar(position="fill", colour="black") +
+  scale_x_discrete("Cluster") +
+  scale_y_continuous("Fraction of cells") +
+  scale_fill_manual(values=sc_all_col_clusters) +
+  AddStyle(title="Clusters in individual HTO", 
+           legend_title="Cluster", 
+           legend_position="right")  
+
+p = patchwork::wrap_plots(list(p1, p2), ncol=2) & theme(legend.position="bottom") 
+p = p + patchwork::plot_annotation("Distribution of HTO and clusters") + 
+  patchwork::plot_layout(guides = "collect")
+p
+
+```
+
+Here it also helps to inspect RNA counts and features in these clusters by hto.
```

Comment:
What do you mean by "in these clusters by hto"?

Reply:
Here it also helps to inspect RNA counts and features per cluster and hto.

Reply:
Done